.changeset/chilled-drinks-travel.md

```diff
@@ -0,0 +1,7 @@
+---
+'@lit-labs/ssr-client': patch
+'lit-html': minor
+'lit': minor
+---
+
+Fix return type of `isTemplateResult` helper to include the `CompiledTemplateResult` and fix the `cache` directive to work correctly with `CompiledTemplateResult`s. Add an explicit error message for ssr hydration to throw on `CompiledTemplate`s. Any usage of `isTemplateResult` without a second argument should also be handling the compiled template.
```

Comment:
I think it would make sense to make 2 separate changeset file, having different lines for `lit-html` and `lit` vs `@lit-labs/ssr-client`.

Reply:
Great suggestion - done